src/wallet/wallet2.cpp

```diff
@@ -3465,6 +3466,18 @@ void wallet2::refresh(bool trusted_daemon, uint64_t start_height, uint64_t & blo
           throw std::runtime_error("proxy exception in refresh thread");
       }
 
+      if (!next_blocks.empty() && !m_blockchain.is_in_bounds(next_blocks_start_height))
```

Comment:
A reorg of more than the threshold could still have the height be in the hash chain bounds. Since this can be used to defend against daemons trolling wallets, that check should probably be removed.

Reply:
Fixed